dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/SerializableSchedule.java

```diff
@@ -71,123 +84,191 @@ public SerializableSchedule() {
+        }
 
-        if (splitDays.length > 2 || splitDays.length < 1 || splitTimes.length > 2 || splitTimes.length < 1) {
-            throw new IllegalArgumentException("Could not parse days configured in schedule: " + strVal);
+        //Parse ZoneID
+        try {
+            this.timezone = splitDaysTimesZone.length == 3 ? ZoneId.of(splitDaysTimesZone[2]) : ZoneId.systemDefault();
+        } catch (DateTimeException e) {
+            throw new IllegalArgumentException(Tr.formatMessage(tc, "invalidScheduleParse", ZoneId.class, splitDaysTimesZone[2],
+                                                                ZoneId.systemDefault().getDisplayName(TextStyle.SHORT, Locale.getDefault())), e);
         }
+    }
+
+    /**
+     * @return the startup
+     */
+    public boolean isStartup() {
+        return startup;
+    }
+
+    /**
+     * @return the validDays
+     */
+    public List<DayOfWeek> getValidDays() {
+        return validDays;
+    }
+
+    /**
+     * @return the startTime
+     */
+    public LocalTime getStartTime() {
+        return startTime;
+    }
 
-        this.validDays = getValidDays(evaluateDayOfWeek(splitDays[0]), splitDays.length == 2 ? evaluateDayOfWeek(splitDays[1]) : null);
-        this.startTime = evaluateLocalTime(splitTimes[0]);
-        this.endTime = splitTimes.length == 2 ? evaluateLocalTime(splitTimes[1]) : null;
+    /**
+     * @return the endTime
+     */
+    public LocalTime getEndTime() {
+        return endTime;
+    }
+
+    /**
+     * @return the timezone
+     */
+    public ZoneId getTimezone() {
+        return timezone;
     }
 
-    private static List<DayOfWeek> getValidDays(DayOfWeek startDay, DayOfWeek endDay) {
+    private static List<DayOfWeek> evaluateValidDays(DayOfWeek startDay, DayOfWeek endDay) {
         //Handle a single day
         if (endDay == null) {
-            return new ArrayList<DayOfWeek>(Arrays.asList(startDay));
+            return Arrays.asList(startDay);
+        }
+
+        //Need to iterate through days of week. Easier to use position in a circular array [0-6]
+        int startDayPosition = startDay.getValue() - 1; //Adjust since DayOfWeek uses values 1-7
+        int endDayPosition = endDay.getValue() % 7; //Adjust if endDay is Sunday (7) we want to iterate till Monday (0).
+
+        //User has specified all days
+        if (startDayPosition == endDayPosition) {
+            return Arrays.asList(DayOfWeek.values());
         }
 
-        //
         ArrayList<DayOfWeek> result = new ArrayList<>();
-        for (int d = startDay.getValue(); d <= endDay.getValue(); d = d == 7 ? 1 : d + 1) {
-            result.add(DayOfWeek.of(d));
+
+        //Increment through circular array [0-6]
+        for (int day = startDayPosition; day != endDayPosition; day = (day + 1) % 7) {
+            result.add(DayOfWeek.of(day + 1));
```

Comment:
I think your algorithm produces the correct result, but I found the variable names confusing when trying to follow it.  `startDayPosition` is actually the day before the start day.